.pre-commit-config.yaml

```diff
@@ -3,45 +3,19 @@
 # SPDX-License-Identifier: Unlicense
 
 repos:
-  - repo: https://github.com/python/black
-    rev: 24.2.0
-    hooks:
-      - id: black
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.13.2
-    hooks:
-      - id: isort
-        args: ["--profile", "black", "--filter-files"]
-  - repo: https://github.com/fsfe/reuse-tool
-    rev: v1.1.2
-    hooks:
-      - id: reuse
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v4.4.0
     hooks:
       - id: check-yaml
       - id: end-of-file-fixer
       - id: trailing-whitespace
-  - repo: https://github.com/pycqa/pylint
-    rev: v2.17.4
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.1.3
     hooks:
-      - id: pylint
-        name: pylint (library code)
-        types: [python]
-        args:
-          - --disable=consider-using-f-string
-        exclude: "^(docs/|examples/|tests/|setup.py$)"
-      - id: pylint
-        name: pylint (example code)
-        description: Run pylint rules on "examples/*.py" files
-        types: [python]
-        files: "^examples/"
-        args:
-          - --disable=consider-using-f-string,duplicate-code,missing-docstring,invalid-name
-      - id: pylint
-        name: pylint (test code)
-        description: Run pylint rules on "tests/*.py" files
-        types: [python]
-        files: "^tests/"
-        args:
-          - --disable=consider-using-f-string,duplicate-code,missing-docstring,invalid-name,protected-access,redefined-outer-name
+      - id: ruff
+        args: ["--select", "I", "--fix"]
```

Comment:
`--fix` is a good one since it will help with local development, but it might help to add the rules to `pyproject.toml`:  (example: https://github.com/tekktrik/circfirm/blob/ab296f8a33db1ceab0b0e5f1fbeb1dc89923396c/pyproject.toml#L82-L84)

Reply:
All for what ever rules make sense. Was trying to keep things simple for the example